lib/Controller/OAuthApiController.php

```diff
@@ -172,6 +172,7 @@ public function generateToken($grant_type, $code = null,
 				} catch (DoesNotExistException $exception) {
 					\OC::$server->getLogger()->logException($exception, ['app'=>__CLASS__]);
 					return new JSONResponse(['error' => 'invalid_grant'], $statusCode);
+					exit();
```

Comment:
unreached code

Reply:
Sorry, but do not know what's a better way to fix this. When the RefreshToken is ended, the expires field will not be filled correctly by a re-authentication through this catch section.
@DeepDiver1975 Maybe you can have a look on this?